src/quicktype-core/language/Dart.ts

```diff
@@ -0,0 +1,353 @@
+        isStartCharacter
+    );
+}
+
+export class DartRenderer extends ConvenienceRenderer {
+    private _currentFilename: string | undefined;
+    private readonly _gettersAndSettersForPropertyName = new Map<Name, [Name, Name]>();
+
+    constructor(
+        targetLanguage: TargetLanguage,
+        renderContext: RenderContext,
+    ) {
+        super(targetLanguage, renderContext);
+
+    }
+
+    protected forbiddenNamesForGlobalNamespace(): string[] {
+        return keywords;
+    }
+
+    protected forbiddenForObjectProperties(_c: ClassType, _className: Name): ForbiddenWordsInfo {
+        return { names: [], includeGlobalForbidden: true };
+    }
+
+    protected makeNamedTypeNamer(): Namer {
+        return typeNamingFunction;
+    }
+
+    protected namerForObjectProperty(): Namer {
+        return propertyNamingFunction;
+    }
+
+    protected makeUnionMemberNamer(): Namer {
+        return propertyNamingFunction;
+    }
+
+    protected makeEnumCaseNamer(): Namer {
+        return enumCaseNamingFunction;
+    }
+
+    protected unionNeedsName(u: UnionType): boolean {
+        return nullableFromUnion(u) === null;
+    }
+
+    protected namedTypeToNameForTopLevel(type: Type): Type | undefined {
+        // If the top-level type doesn't contain any classes or unions
+        // we have to define a class just for the `FromJson` method, in
+        // emitFromJsonForTopLevel.
+        return directlyReachableSingleNamedType(type);
+    }
+
+    protected makeNamesForPropertyGetterAndSetter(
+        _c: ClassType,
+        _className: Name,
+        _p: ClassProperty,
+        _jsonName: string,
+        name: Name
+    ): [Name, Name] {
+        const getterName = new DependencyName(propertyNamingFunction, name.order, lookup => `get_${lookup(name)}`);
+        const setterName = new DependencyName(propertyNamingFunction, name.order, lookup => `set_${lookup(name)}`);
+        return [getterName, setterName];
+    }
+
+    protected makePropertyDependencyNames(
+        c: ClassType,
+        className: Name,
+        p: ClassProperty,
+        jsonName: string,
+        name: Name
+    ): Name[] {
+        const getterAndSetterNames = this.makeNamesForPropertyGetterAndSetter(c, className, p, jsonName, name);
+        this._gettersAndSettersForPropertyName.set(name, getterAndSetterNames);
+        return getterAndSetterNames;
+    }
+
+    protected startFile(basename: Sourcelike): void {
+        assert(this._currentFilename === undefined, "Previous file wasn't finished");
+        // FIXME: The filenames should actually be Sourcelikes, too
+        this._currentFilename = `${this.sourcelikeToString(basename)}.dart`;
+    }
+
+    protected finishFile(): void {
+        super.finishFile(defined(this._currentFilename));
+        this._currentFilename = undefined;
+    }
+
+    protected emitFileHeader(fileName: Sourcelike): void {
+        this.startFile(fileName);
+        this.ensureBlankLine();
+        this.ensureBlankLine();
```

Comment:
`ensureBlankLine` is idempotent - a second invocation won't do anything.  I actually need two blank lines myself now in Python, so I suggest you just add a FIXME and I'll handle it when I've implemented it.

Reply:
It's an extra line, not needed. I will clean it up.